### PR TITLE
Update citation information.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Adds new citation information for the SciPy 2021 proceedings paper (no DOI is available yet). For now, I recommend citing the new SciPy paper only in cases where the content specifically references the new features that have been recently added:

> If your paper makes use of the signac-flow features for aggregation, groups, and bundling or signac features for synced collections, consider citing:

I reorganized citation information to use BibTeX files that match the format of the "Scientific Papers" page.

I made the recommended citation statement shorter, since I have seen several users (including myself) adapt the statement in this way.

Lastly, I edited some bibtex-tidy pre-commit settings to wrap content at 90 characters. That's what fits on screen (on a desktop or laptop) without creating a scroll bar.

Compare the ReadTheDocs build to the current page: https://docs.signac.io/en/latest/acknowledge.html

## Motivation and Context
Updating citation information to make it clearer how to cite the software.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

File: docs/source/acknowledge.rst [hunk 1]
```diff
@@ -17,62 +17,49 @@ Preferred Citation
...
+If you make use of **signac-flow**, we also strongly recommend citing the following publication:
+
+.. bibliography::
+    :filter: False
+
+    signac_scipy_2018
+
+--------
+
+If your paper makes use of the **signac-flow** features for aggregation, groups, and bundling or **signac** features for synced collections, consider citing:
```
Comment (csadorf) [inline docs/source/acknowledge.rst:40]: I would remove bundling from that. This feature has been around for a while.

File: docs/source/acknowledge.rst [hunk 2]
```diff
@@ -17,62 +17,49 @@ Preferred Citation
...
-                      Sharon C. Glotzer},
-      title        = {glotzerlab/signac},
-      month        = feb,
-      year         = 2019,
-      doi          = {10.5281/zenodo.2581327},
-      url          = {https://doi.org/10.5281/zenodo.2581327}
-    }
+
+.. literalinclude:: acknowledge.bib
+    :language: bibtex
```
Comment (csadorf) [inline docs/source/acknowledge.rst:62]: I'd recommend to leave the easy copy & paste option. For some reason I believe that it makes it more likely for people to cite. Citations are extremely valuable for some of us and we should reduce any perceived barrier if possible. 

If you think it's getting too large, maybe just show it for CompMat and add JS buttons to copy the other ones into the clipboard.

File: docs/source/acknowledge.rst (hunk 1 above)
Comment (bdice) [reply docs/source/acknowledge.rst:40]: Good idea.

File: docs/source/acknowledge.rst (hunk 2 above)
Comment (bdice) [reply docs/source/acknowledge.rst:62]: I agree that this needs to be easy. I am not sure what you mean about the changes. If you mean copy-paste BibTeX, this is nearly equivalent to the existing page when rendered — look at what it does when rendered and let me know if it addresses your question here?

Comment (csadorf) [reply docs/source/acknowledge.rst:62]: Ah sorry, I was reviewing on my phone and thought you had replaced it with a download link. All good.

**Review by csadorf — APPROVED** [review]

**Review by cbkerr — APPROVED** [review]
I couldn't find a RTD-built preview at https://readthedocs.org/projects/signac/builds/14172973/ but the diffs look good to me.

**Pull request merged**